generator_test.go

```diff
@@ -175,6 +190,33 @@ func testNewV1MissingNetworkFaultyRand(t *testing.T) {
 	}
 }
 
+func testNewV1MissingNetworkFaultyRandWithOptions(t *testing.T) {
+	g := NewGenWithOptions(
+		WithHWAddrFunc(func() (net.HardwareAddr, error) {
+			return []byte{}, fmt.Errorf("uuid: no hw address found")
+		}),
+		WithRandomReader(&faultyReader{
+			readToFail: 1,
+		}),
+	)
+
+	u, err := g.NewV1()
```

Comment:
any value in checking the error message?

Reply:
I can't imagine what else to check at this point. I just kept the same behaviour as the existing tests, I didn't want to modify anything pre-existing.